consensus/src/core.rs

```diff
@@ -440,24 +469,143 @@ impl Core {
+                                    warn!("Fast Sync: Ignoring non-checkpoint message {:?} during sync phase", other_message);
+                                }
+                                None => { // Channel closed
+                                     warn!("Fast Sync: Message channel closed unexpectedly. Starting from Genesis.");
+                                     break; // Exit the response waiting loop
+                                }
+                            }
+                        },
+
+                        // Handle timeout
+                        () = &mut deadline => {
+                            warn!("Fast Sync: Timeout waiting for checkpoint response. Starting from Genesis.");
+                            break; // Exit the response waiting loop
+                        },
+                    }
+                } // End of response waiting loop
+
+            } else {
+                warn!("Fast Sync: No peers found to request checkpoint from. Starting from Genesis.");
+            }
+
+            // If sync failed, core state remains at Genesis defaults. If succeeded, it's updated.
+             if checkpoint_fetched {
+                 info!("Fast Sync: Successfully initialized state from checkpoint.");
+             } else {
+                 info!("Fast Sync: Failed to obtain valid checkpoint, proceeding with Genesis state.");
+             }
+
+        } // End of #[cfg(feature = "fast-sync")] block
+
+
+        // --- Normal Startup / Post-Sync ---
+        // Initialize timer and potentially generate first block based on the current state
+        // (which might be Genesis or from fast sync)
         self.timer.reset();
+        info!("Core starting normal operation at Epoch {}, Round {}", self.current_epoch, self.round);
         if self.name == self.leader_elector.get_leader(self.round) {
+            // generate_proposal uses self.round, self.current_epoch, self.high_qc which are now set correctly.
             self.generate_proposal(None).await;
         }
 
-        // This is the main loop: it processes incoming blocks and votes,
-        // and receive timeout notifications from our Timeout Manager.
+        // --- Main Loop ---
         loop {
             let result = tokio::select! {
-                Some(message) = self.rx_message.recv() => match message {
-                    ConsensusMessage::Propose(block) => self.handle_proposal(&block).await,
-                    ConsensusMessage::Vote(vote) => self.handle_vote(&vote).await,
-                    ConsensusMessage::Timeout(timeout) => self.handle_timeout(&timeout).await,
-                    ConsensusMessage::TC(tc) => self.handle_tc(tc).await,
-                    _ => panic!("Unexpected protocol message")
+                Some(message) = self.rx_message.recv() => {
+                    // Add guards to ignore sync messages during normal operation
```

Comment:
Why does there have to be normal & "fast sync" operation?

E.g. what if `Alice` runs a _normal_ node, and `Bob` runs fast sync - it should be fine for `Alice` to reply to `Bob`'s requests - or not?

Reply:
Looking at it now, I understand your confusion.
Initially, I wanted to make sure I was separating fast-sync nodes logic from "normal" nodes as my fabric run wasn't yielding the expected result.
After implementing the feature flag this wasn't a concern anymore, but this leftover definitely needs some cleanup.
To answer your question, sync modes aren't mutually exclusive on a network communication level.